

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>5.User Guide &mdash; Octopus-toolkit stable documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="Octopus-toolkit stable documentation" href="index.html"/>
        <link rel="next" title="6.Tutorial" href="tutorial.html"/>
        <link rel="prev" title="4.User Interface" href="user_interface.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Octopus-toolkit
          

          
          </a>

          
            
            
              <div class="version">
                stable
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="quick_start.html">0.Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">1.Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="run.html">2.Run</a></li>
<li class="toctree-l1"><a class="reference internal" href="folder.html">3.Octopus-toolkit folder</a></li>
<li class="toctree-l1"><a class="reference internal" href="user_interface.html">4.User Interface</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">5.User Guide</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#rd-party-tools-used-in-octopus-toolkit">5-1.3rd party tools used in Octopus-toolkit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#public-data">5-2.Public data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#quick-start">Quick Start</a></li>
<li class="toctree-l3"><a class="reference internal" href="#work-flow">Work flow</a></li>
<li class="toctree-l3"><a class="reference internal" href="#start-analyzing-public-data">Start analyzing public data</a></li>
<li class="toctree-l3"><a class="reference internal" href="#octopus-option">Octopus option</a></li>
<li class="toctree-l3"><a class="reference internal" href="#run">Run</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#private-data">5-3.Private data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id2">Quick Start</a></li>
<li class="toctree-l3"><a class="reference internal" href="#start-analyzing-private-data">Start analyzing private data</a></li>
<li class="toctree-l3"><a class="reference internal" href="#private-table">Private table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id3">Octopus option</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id4">Run</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#peak-calling">5-4.Peak Calling</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id5">Quick Start</a></li>
<li class="toctree-l3"><a class="reference internal" href="#start-analyzing-peak-calling">Start analyzing peak calling</a></li>
<li class="toctree-l3"><a class="reference internal" href="#peak-calling-table">Peak calling table</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#graph">5-5.Graph</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id6">Quick Start</a></li>
<li class="toctree-l3"><a class="reference internal" href="#start-analyzing-graph">Start analyzing Graph</a></li>
<li class="toctree-l3"><a class="reference internal" href="#graph-table">Graph table</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#visualization">5-6.Visualization</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id7">Quick Start</a></li>
<li class="toctree-l3"><a class="reference internal" href="#start-analyzing-igv">Start analyzing IGV</a></li>
<li class="toctree-l3"><a class="reference internal" href="#igv-table">IGV table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id8">Run</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output">5-7.Output</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#fastq">00_Fastq</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#full-parameter">5-8.Full Parameter</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#preprocessing">Preprocessing</a></li>
<li class="toctree-l3"><a class="reference internal" href="#qc-trimming">QC &amp; Trimming</a></li>
<li class="toctree-l3"><a class="reference internal" href="#alignment">Alignment</a></li>
<li class="toctree-l3"><a class="reference internal" href="#visualization-tagdirectory">Visualization-TagDirectory</a></li>
<li class="toctree-l3"><a class="reference internal" href="#visualization-makebigwig">Visualization-MakeBigWig</a></li>
<li class="toctree-l3"><a class="reference internal" href="#peakcalling-chip-seq-histone">PeakCalling-ChIP-Seq/Histone</a></li>
<li class="toctree-l3"><a class="reference internal" href="#peakcalling-peak-filter">PeakCalling-Peak Filter</a></li>
<li class="toctree-l3"><a class="reference internal" href="#peakcalling-other-analysis">PeakCalling-Other analysis</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">6.Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="error.html">7.Error Code</a></li>
<li class="toctree-l1"><a class="reference internal" href="license.html">8.License</a></li>
<li class="toctree-l1"><a class="reference internal" href="help.html">9.Help</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">Octopus-toolkit</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>5.User Guide</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/guide.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="user-guide">
<h1>5.User Guide<a class="headerlink" href="#user-guide" title="Permalink to this headline">¶</a></h1>
<p>Octopus-toolkit can analyze data already published on NCBI or stored on your computer.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Public</span> <span class="pre">data</span></code> : The NGS data uploaded to NCBI.</li>
<li><code class="docutils literal"><span class="pre">Private</span> <span class="pre">data</span></code> : The NGS data stored on the user&#8217;s computer. (Data obtained through a sequencing machine or downloaded directly in the web)</li>
</ul>
<p>Also, Octopus-toolkit can analyze further analysis such as <code class="docutils literal"><span class="pre">Peak</span> <span class="pre">Calling</span></code>, <code class="docutils literal"><span class="pre">Drawing</span> <span class="pre">Graph</span></code>, and <code class="docutils literal"><span class="pre">Visualization</span></code> after analyzing the public data or private data.</p>
<p>The basic analysis of Octopus-toolkit uses public data.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">NGS</span> <span class="pre">data</span> <span class="pre">processing</span></code></li>
</ul>
<img alt="_images/0.NGS_data_processing.png" src="_images/0.NGS_data_processing.png" />
<p>The analysis using Octopus-toolkit is based on the process from preprocessing to visualization. You can analyze the further analysis after the completion of the visualization process.</p>
<div class="section" id="rd-party-tools-used-in-octopus-toolkit">
<span id="rd-party-tool"></span><h2>5-1.3rd party tools used in Octopus-toolkit<a class="headerlink" href="#rd-party-tools-used-in-octopus-toolkit" title="Permalink to this headline">¶</a></h2>
<p>Octopus-toolkit uses the following 3rd party tools to analyze NGS data.</p>
<table border="1" class="colwidths-given docutils">
<colgroup>
<col width="25%" />
<col width="25%" />
<col width="25%" />
<col width="25%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">NGS Process</th>
<th class="head">Function</th>
<th class="head">3rd party tool</th>
<th class="head">Sub-tool</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Preprocessing</span></code></td>
<td>Download SRA file from NCBI</td>
<td><code class="docutils literal"><span class="pre">Aspera</span></code></td>
<td>ascp</td>
</tr>
<tr class="row-odd"><td>&nbsp;</td>
<td>Convert SRA to Fastq</td>
<td><code class="docutils literal"><span class="pre">SRAToolkit</span></code></td>
<td>fastq-dump</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Quality</span> <span class="pre">check</span></code></td>
<td>Quality check for raw data</td>
<td><code class="docutils literal"><span class="pre">FastQC</span></code></td>
<td>fastqc</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Trimming</span></code></td>
<td>Trimming for adapter sequence and low-quality reads</td>
<td><code class="docutils literal"><span class="pre">Trimmomatic</span></code></td>
<td>&nbsp;</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Alignment</span></code></td>
<td>Indexing a reference genome</td>
<td><code class="docutils literal"><span class="pre">Hisat2</span></code></td>
<td>hisat2-build</td>
</tr>
<tr class="row-odd"><td>&nbsp;</td>
<td>Mapping the reads to the reference genome</td>
<td><code class="docutils literal"><span class="pre">Hisat2</span></code></td>
<td>hisat2-align</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Visualization</span></code></td>
<td>Create a visualization file</td>
<td><code class="docutils literal"><span class="pre">Homer</span></code></td>
<td>makeTagDirectory,makeUCSCFile,analyzeRepeats</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Peak</span> <span class="pre">calling</span></code></td>
<td>Detect mapped read to genome</td>
<td><code class="docutils literal"><span class="pre">Homer</span></code></td>
<td>findPeak, pos2bed,annotatePeaks</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Graph</span></code></td>
<td>Calculate the value of bigWig for a specific region</td>
<td><code class="docutils literal"><span class="pre">Bwtool</span></code></td>
<td>matrix</td>
</tr>
<tr class="row-odd"><td>&nbsp;</td>
<td>Draw the heatmap and line plot</td>
<td><code class="docutils literal"><span class="pre">R</span></code></td>
<td>pheatmap, ggplot2</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">IGV</span></code></td>
<td>Use IGV tool for visualization</td>
<td><code class="docutils literal"><span class="pre">IGV</span></code></td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="public-data">
<h2>5-2.Public data<a class="headerlink" href="#public-data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="quick-start">
<h3>Quick Start<a class="headerlink" href="#quick-start" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>Enter the GEO accession number or click the open button to load GEO accession number list.</li>
<li>Click the Run button.</li>
<li>Choose the option you want to use in the Octopus Option.</li>
<li>Click the Run button in the Octopus Option.</li>
</ol>
</div>
</div>
<div class="section" id="work-flow">
<h3>Work flow<a class="headerlink" href="#work-flow" title="Permalink to this headline">¶</a></h3>
<img alt="_images/13.analysis_workflow2.png" src="_images/13.analysis_workflow2.png" />
</div>
<div class="section" id="start-analyzing-public-data">
<h3>Start analyzing public data<a class="headerlink" href="#start-analyzing-public-data" title="Permalink to this headline">¶</a></h3>
<img alt="_images/1.Public_Data_Input.png" src="_images/1.Public_Data_Input.png" />
<p>To analyze the public data, enter the GEO (Gene Expression Omnibus) Accession number or load the GEO Accession number list file.</p>
<ul>
<li><p class="first">Input : <code class="docutils literal"><span class="pre">GEO</span> <span class="pre">Accession</span> <span class="pre">number</span></code></p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">GSExxx</span> <span class="p">:</span> <span class="n">Each</span> <span class="n">Series</span><span class="p">(</span><span class="n">Study</span><span class="p">)</span> <span class="n">record</span> <span class="ow">is</span> <span class="n">assigned</span> <span class="n">a</span> <span class="n">unique</span> <span class="ow">and</span> <span class="n">stable</span> <span class="n">GEO</span> <span class="n">accession</span> <span class="n">number</span><span class="o">.</span>
<span class="n">GSMxxx</span> <span class="p">:</span> <span class="n">Each</span> <span class="n">Sample</span> <span class="n">record</span> <span class="ow">is</span> <span class="n">assigned</span> <span class="n">a</span> <span class="n">unique</span> <span class="ow">and</span> <span class="n">stable</span> <span class="n">GEO</span> <span class="n">accession</span> <span class="n">number</span><span class="o">.</span>
</pre></div>
</div>
</li>
<li><p class="first">Input : <code class="docutils literal"><span class="pre">GEO</span> <span class="pre">Accession</span> <span class="pre">number</span> <span class="pre">list</span></code> (<a class="reference download internal" href="_downloads/GEO_Accession_number.list" download=""><code class="xref download docutils literal"><span class="pre">example.list</span></code></a>)</p>
</li>
</ul>
<ol class="arabic simple">
<li>Enter the GEO accession number or click the open button to load GEO accession number list.</li>
<li>Click the Run button.</li>
</ol>
</div>
<div class="section" id="octopus-option">
<span id="id1"></span><h3>Octopus option<a class="headerlink" href="#octopus-option" title="Permalink to this headline">¶</a></h3>
<p>You can use the options provided by Octopus-toolkit as well as the tools used by each process.</p>
<img alt="_images/2.Octopus_Option.png" src="_images/2.Octopus_Option.png" />
<table border="1" class="colwidths-given docutils">
<colgroup>
<col width="22%" />
<col width="78%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Option</th>
<th class="head">Decription</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Latest</span> <span class="pre">genome</span> <span class="pre">version</span></code></td>
<td>Use the latest genome.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Analyze</span> <span class="pre">the</span> <span class="pre">data</span> <span class="pre">in</span> <span class="pre">succession</span></code></td>
<td>Skips already analyzed the samples.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Omit</span> <span class="pre">Process</span></code></td>
<td>Omit the selected process. (Trimming and Sort)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">CPU(Thread)</span></code></td>
<td>Set the CPU to use.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Strand(RNA)</span></code></td>
<td>Set the library strand for RNA-Seq.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Use</span> <span class="pre">the</span> <span class="pre">full</span> <span class="pre">parameter</span> <span class="pre">for</span> <span class="pre">each</span> <span class="pre">tool</span></code></td>
<td>Use the full parameter.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Edit</span> <span class="pre">Button</span></code></td>
<td>Go to the Full parameter option window.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Remove</span> <span class="pre">Files</span></code></td>
<td>Delete selected files after each process analysis for HDD free space.</td>
</tr>
</tbody>
</table>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Latest</span> <span class="pre">genome</span> <span class="pre">version</span></code></li>
</ul>
<p>Octopus-toolkit can analyze for Homo sapiens, Mus musculus, Drosophila melanogaster, Saccharomyces cerevisiae, and Canis lupus familaris.</p>
<table border="1" class="colwidths-given docutils" id="id11">
<caption><span class="caption-text">Available analysis genome version</span><a class="headerlink" href="#id11" title="Permalink to this table">¶</a></caption>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Organism</th>
<th class="head">Genome version</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Homo sapiens</td>
<td>hg38, hg19, hg18</td>
</tr>
<tr class="row-odd"><td>Mus musculus</td>
<td>mm10, mm9</td>
</tr>
<tr class="row-even"><td>Drosophila melanogaster</td>
<td>dm6, dm3</td>
</tr>
<tr class="row-odd"><td>Saccharomyces cerevisiae</td>
<td>sacCer3</td>
</tr>
<tr class="row-even"><td>Canis lupus familaris</td>
<td>canFam3</td>
</tr>
</tbody>
</table>
<p>The latest genome version uses the latest version of the genome for analysis. If you don&#8217;t select this option, Octopus-toolkit uses the following genome version for analysis.</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="o">*</span> <span class="n">Latest</span> <span class="n">genome</span> <span class="p">(</span><span class="n">O</span><span class="p">)</span> <span class="p">:</span> <span class="n">hg38</span><span class="p">,</span> <span class="n">mm10</span><span class="p">,</span> <span class="n">dm6</span><span class="p">,</span> <span class="n">sacCer3</span><span class="p">,</span> <span class="n">canFam3</span>
<span class="o">*</span> <span class="n">Latest</span> <span class="n">genome</span> <span class="p">(</span><span class="n">X</span><span class="p">)</span> <span class="p">:</span> <span class="n">hg19</span><span class="p">,</span> <span class="n">mm9</span><span class="p">,</span> <span class="n">dm3</span><span class="p">,</span> <span class="n">sacCer3</span><span class="p">,</span> <span class="n">canFam3</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Analyze</span> <span class="pre">the</span> <span class="pre">data</span> <span class="pre">in</span> <span class="pre">succession</span></code></li>
</ul>
<p>When analyzing a GSE accession number that has a lot of sample data, you may have to shut down the computer during analysis for a variety of reasons. For example, if you stop analyzing after 8 samples of 10 samples have been analyzed, and later analyze again, you will have to analyze again from the beginning.(First sample)</p>
<p>To solve this issue, Octopus-toolkit saves analysis information of the sample in the log file after the analysis of one sample is completed.</p>
<p>If you select analyze the data in succession, you can skip the already analyzed samples.</p>
<p>If you have already analyzed the sample, but you want to reanalyze it with other options, Don&#8217;t select analyze the data in succession.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Omit</span> <span class="pre">process</span></code></li>
</ul>
<p>The omit process allows you to skip and analyze the selected process during the entire analysis process. You can shorten the overall analysis time by omitting these processes.</p>
<p>In the trimming process, If all reads have bad quality, there is a possibility that all reads are deleted. Octopus-toolkit will analyze the original raw data(Fastq) when all reads are deleted due to bad quality after trimming.</p>
<p>Also, If all reads are higher than the trimming cutoff (quality, adapt), you can skip to the
trimming process.</p>
<p>In the sorting process, The alignment process in Octopus-toolkit provides you with a bam format file and a sorted.bam file that sorts it.</p>
<p>Sorted.bam files can&#8217;t be used for visualization , peak calling, and graph function after alignment process. It is used only when confirming the bam file directly via IGV tool.
(The IGV process basically uses the bigWig format file, which is the output file of Visualization)</p>
<p>However, the reason for creating sorted.bam file in Octopus-toolkit is that many NGS tools use the sorted.bam file as input for further analysis.</p>
<p>So, If you don&#8217;t need a sorted.bam file you can skip this sorting process to save time.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">CPU(Thread)</span></code></li>
</ul>
<p>Octopus-toolkit can select the CPU to your computer for analysis. (Default : 8)</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Strand(RNA)</span></code></li>
</ul>
<p>Strand is the library&#8217;s strand information needed to analyze RNA-Seq data.</p>
<p>Octopus-toolkit extracts information from samples in the GEO dataset when analyzing public data. However, information about strand is not well provided in GEO datasets.</p>
<p>So, when you want to analyze the RNA-Seq data, you can set the strand specific library or non-strand library via this option. (Default : Unstrand)</p>
<p>You can select either non-strand library or the strand-specific library such as FR-Firststrand, FR-Secondstrand using this option.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Use</span> <span class="pre">the</span> <span class="pre">full</span> <span class="pre">parameter</span> <span class="pre">for</span> <span class="pre">each</span> <span class="pre">tool</span></code></li>
</ul>
<p>Use the full parameter for each tool (Checkbox) allows you to select the full option of the tool used in each analysis.</p>
<p>When you select it, the Edit button is enabled. You can go to the Full parameter option window via the Edit button.</p>
<p>Follow the link for full parameter details : <a class="reference internal" href="#full-parameter"><span class="std std-ref">Full Parameter</span></a></p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Edit</span> <span class="pre">Button</span></code></li>
</ul>
<p>The Edit button will be active when you select Use the full parameter for each tool. If you click the Edit button, the Full parameter optinos window will appear.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Remove</span> <span class="pre">Files</span></code></li>
</ul>
<p>The data generated in the NGS analysis process has capacity over KiloByte to GigaByte, and your computer needs a lot of free space to store it. However, all the generated data is not needed for you.
The generated data except for the main data for further analysis is deleted after each process by the your selection.</p>
<table border="1" class="colwidths-given docutils">
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Option</th>
<th class="head">Extension</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">SRA</span></code></td>
<td>sra</td>
<td>Compressed raw data downloaded from NCBI. (Sequence Read Archive)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Fastq</span></code></td>
<td>fastq or fq</td>
<td>Raw data converted during preprocessing. (A short read sequence file)</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Fastqc</span></code></td>
<td>html or text</td>
<td>Output generated during Quality Check. (output of FastQC)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Trimming</span></code></td>
<td>fastq or fq</td>
<td>Output generated during Trimming. (Trimmed raw file(Fastq))</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">BAM</span></code></td>
<td>bam</td>
<td>Output generated during Alignment. (Mapped read to the genome)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Sorted_Bam</span></code></td>
<td>bam</td>
<td>Output generated during Sortring. (Sorted mapped read)</td>
</tr>
</tbody>
</table>
<ol class="arabic simple" start="3">
<li>Choose the option you want to use in the Octopus Option.</li>
<li>Click the Run button in the Octopus Option.</li>
</ol>
</div>
<div class="section" id="run">
<h3>Run<a class="headerlink" href="#run" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>Analysis screen of Octopus-toolkit (Public data analysis)</li>
</ul>
<img alt="_images/3.Octopus-toolkit_run_public.png" src="_images/3.Octopus-toolkit_run_public.png" />
</div>
</div>
<div class="section" id="private-data">
<h2>5-3.Private data<a class="headerlink" href="#private-data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id2">
<h3>Quick Start<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>Select Analysis -&gt; Private Data in the Menu bar.</li>
<li>Select the folder where the raw data is stored or select raw data stored in your computer.</li>
<li>Add information about each sample in the private table.</li>
<li>Click the Run button in the private table.</li>
<li>choose the option you want to use in the Octopus option.</li>
<li>Click the Run button in the Octopus Option.</li>
</ol>
</div>
</div>
<div class="section" id="start-analyzing-private-data">
<h3>Start analyzing private data<a class="headerlink" href="#start-analyzing-private-data" title="Permalink to this headline">¶</a></h3>
<img alt="_images/4.Private_Start.png" src="_images/4.Private_Start.png" />
<p>Unlike public data analysis, private data analysis does not download SRA file from NCBI. This analysisonly uses the raw data stored on your computer.</p>
<p>Private data analysis uses the Fastq format file and the gzip-compressed Fastq.gz format file as input.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Raw data : Sample <code class="docutils literal"><span class="pre">.fastq</span></code> or Sample <code class="docutils literal"><span class="pre">.fq</span></code></li>
<li>compressed Raw data : Sample <code class="docutils literal"><span class="pre">.fastq.gz</span></code> or Sample <code class="docutils literal"><span class="pre">.fq.gz</span></code></li>
<li>Single-End data : Sample <code class="docutils literal"><span class="pre">.fastq</span></code> (or fq, fastq.gz, fq.gz)</li>
<li>Paired-End data : Sample <code class="docutils literal"><span class="pre">_1.fastq</span></code>, Sample <code class="docutils literal"><span class="pre">_2.fastq</span></code></li>
</ul>
</div>
<p>For Paired-End data, distinguish between <code class="docutils literal"><span class="pre">_1.fastq</span></code> and <code class="docutils literal"><span class="pre">_2.fastq</span></code> in the same sample name.</p>
<ol class="arabic simple">
<li>To analyze the private data, select <code class="docutils literal"><span class="pre">Analysis</span></code> -&gt; <code class="docutils literal"><span class="pre">Private</span> <span class="pre">Data</span></code> in the menu bar.</li>
<li>Select the <code class="docutils literal"><span class="pre">folder</span></code> where the raw data is stored or select <code class="docutils literal"><span class="pre">raw</span> <span class="pre">data</span></code> stored in your computer.</li>
</ol>
<p>Octopus-toolkit examines the selected raw data or folder and selects only the data that matches the input format and outputs it to the private table.</p>
</div>
<div class="section" id="private-table">
<h3>Private table<a class="headerlink" href="#private-table" title="Permalink to this headline">¶</a></h3>
<img alt="_images/5.Private_Table.png" src="_images/5.Private_Table.png" />
<p>Octopus-toolkit does not know the sample information about your private data. So you need to add information about the samples you want to analyze.</p>
<p>Octopus-toolkit checks the name of the selected raw data to create an initial table. Paired-End data creates a table by separating <code class="docutils literal"><span class="pre">_1.fastq</span></code> and <code class="docutils literal"><span class="pre">_2.fastq</span></code> format from data of the <code class="docutils literal"><span class="pre">same</span> <span class="pre">name</span></code>.</p>
<p>If the selected data is not displayed in the table, refer to the data format of the above note and change the data name.</p>
<p>First, select a sample in the table to enter information. Insert information about the selected sample by selecting it from the table option below.</p>
<table border="1" class="colwidths-given docutils">
<colgroup>
<col width="33%" />
<col width="67%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Option</th>
<th class="head">Decription</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Genome</span></code></td>
<td>Select the sample&#8217;s genome.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Seq</span> <span class="pre">type</span></code></td>
<td>Select the sample&#8217;s seq type.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Multi-Lane</span></code></td>
<td>Select samples sequenced by Multi-Lane.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Strand</span></code></td>
<td>Select the sample&#8217;s strand.</td>
</tr>
</tbody>
</table>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Genome</span></code></li>
</ul>
<p>The available genome in Octopus-toolkit is:</p>
<table border="1" class="colwidths-given docutils">
<colgroup>
<col width="44%" />
<col width="56%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Species</th>
<th class="head">Genome version</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Homo</span> <span class="pre">sapiens</span></code></td>
<td><code class="docutils literal"><span class="pre">hg38</span></code> (Dec.2013, GRCh38), <code class="docutils literal"><span class="pre">hg19</span></code> (Feb.2009,GRCh37), <code class="docutils literal"><span class="pre">hg18</span></code> (Mar.2006 NCBI36)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Mus</span> <span class="pre">musculus</span></code></td>
<td><code class="docutils literal"><span class="pre">mm10</span></code> (Dec.2011 GRCm38), <code class="docutils literal"><span class="pre">mm9</span></code> (July.2007 NCBI37)</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Drosophila</span> <span class="pre">melanogaster</span></code></td>
<td><code class="docutils literal"><span class="pre">dm6</span></code> (Aug.2014 BDGP Release 6+ ISO1 MT), <code class="docutils literal"><span class="pre">dm3</span></code> (Apr.2006 BDGP R5)</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Saccharomyces</span> <span class="pre">cerevisiae</span></code></td>
<td><code class="docutils literal"><span class="pre">sacCer3</span></code> (Apr.2011 SacCer_Apr2011)</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Canis</span> <span class="pre">lupus</span> <span class="pre">familaris</span></code></td>
<td><code class="docutils literal"><span class="pre">canFam3</span></code> (Sep.2011 Broad CanFam3.1)</td>
</tr>
</tbody>
</table>
<ul class="simple" id="seq-type">
<li><code class="docutils literal"><span class="pre">Seq</span> <span class="pre">type</span></code></li>
</ul>
<p>Octopus-toolkit can analyze ChIP-Seq, RNA-Seq, MeDIP-Seq, ATAC-Seq, Dnase-Seq and Mnase-Seq.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Multi-Lane</span></code></li>
</ul>
<p>When sequencing through a sequencing machine, Multi-Lane means that raw data is extracted by dividing into several lanes rather than 1 lane.</p>
<p>Most multi-lane data have the following filenames.</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">Sample</span><span class="o">.</span><span class="n">L001</span><span class="o">.</span><span class="n">fastq</span><span class="p">,</span> <span class="n">Sample</span><span class="o">.</span><span class="n">L002</span><span class="o">.</span><span class="n">fastq</span><span class="p">,</span> <span class="n">Sample</span><span class="o">.</span><span class="n">L003</span><span class="o">.</span><span class="n">fastq</span> <span class="o">...</span> <span class="n">Sample</span><span class="o">.</span><span class="n">L008</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
<p>To set multi-lane information, all multi-lane sample data should be selected with the same number of multi-lane values.</p>
<p>If you want to know more information, please refer to the tutorial site. Multi-Lane tutorial Link</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Strand</span></code></li>
</ul>
<p>The strand provided by Octopus-toolkit is a library strand of two types.</p>
<ol class="arabic simple">
<li>Non-strand library : <code class="docutils literal"><span class="pre">Unstrand</span></code> (Default)</li>
<li>Strand specific library : <code class="docutils literal"><span class="pre">FR-Firststrand</span></code>, <code class="docutils literal"><span class="pre">FR-secondstrand</span></code></li>
</ol>
<p>To add this option, the Seq type of the sample to be selected is RNA-Seq.</p>
<ol class="arabic simple" start="3">
<li>Add information about each sample in the private table.</li>
<li>Click the Run button in the private table.</li>
</ol>
</div>
<div class="section" id="id3">
<h3>Octopus option<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<p>Octopus option is the same as public data analysis. Please refer to public data analysis. (<a class="reference internal" href="#octopus-option"><span class="std std-ref">Octopus option</span></a>)</p>
</div>
<div class="section" id="id4">
<h3>Run<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>Analysis screen of Octopus-toolkit (Private data analysis)</li>
</ul>
<img alt="_images/6.Octopus-toolkit_run_private.png" src="_images/6.Octopus-toolkit_run_private.png" />
</div>
</div>
<div class="section" id="peak-calling">
<h2>5-4.Peak Calling<a class="headerlink" href="#peak-calling" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id5">
<h3>Quick Start<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>Select Analysis -&gt; Peak Calling in the Menu bar.</li>
<li>Select the resulting folder generated by analyzing public data or private data.</li>
<li>Add information about each sample in the peak calling table.</li>
<li>Click the Run button in the peak calling table.</li>
</ol>
</div>
</div>
<div class="section" id="start-analyzing-peak-calling">
<h3>Start analyzing peak calling<a class="headerlink" href="#start-analyzing-peak-calling" title="Permalink to this headline">¶</a></h3>
<img alt="_images/7.Peak_Calling_Start.png" src="_images/7.Peak_Calling_Start.png" />
<p>The peak calling is the process of identifying the region of the mapped read to the genome.
Peak calling process of Octopus-toolkit can&#8217;t perform RNA-Seq analysis in NGS-Seq.</p>
<p>Peak calling will detect the region of the read mapped on the genome, so you need to complete the alignment process in advance. Therefore, in order to use peak calling, you must complete at least one sample of the data analysis. (Public or Private)</p>
<ol class="arabic simple">
<li>Select Analysis -&gt; Peak Calling in the Menu bar.</li>
<li>Select the resulting folder generated by analyzing public data or private data.</li>
</ol>
</div>
<div class="section" id="peak-calling-table">
<h3>Peak calling table<a class="headerlink" href="#peak-calling-table" title="Permalink to this headline">¶</a></h3>
<img alt="_images/8.Peak_Calling_Table.png" src="_images/8.Peak_Calling_Table.png" />
<p>The Peak calling table consists of a table that stores sample information, a sample input window, and a table option window.</p>
<p>To use the peak calling, select the sample you want to analyze in the sample window and click the insert button. And then, Add Control and Style information for the selected sample After selecting the inserted sample.</p>
<p>If Control data associated with the sample exist in the analyzed result, Octopus-toolkit will not automatically identify the Control data, so you will need to add Control information for the sample.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Control</span></code></li>
</ul>
<p>To use Input or IgG seqeuencing related to sample, as a control, this option allows you to select Input or IgG sequencing. (Recommended)</p>
<p>Control information can be selected only for the sample included in the resulting folder.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Style</span></code></li>
</ul>
<p>Peak calling process has specific analysis method about seq type of each sample.</p>
<table border="1" class="colwidths-given docutils">
<colgroup>
<col width="25%" />
<col width="25%" />
<col width="50%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">option</th>
<th class="head">Seq type</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Transcription</span> <span class="pre">Factor</span></code></td>
<td>ChIP-Seq, DNase-Seq</td>
<td>Peak finding for single contact or focal ChIP-Seq experiments or DNase-Seq.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">Histone</span></code></td>
<td>ChIP-Seq</td>
<td>Peak finding for broad regions of enrichment found in ChIP-Seq experiments for various histone marks.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">Groseq</span></code></td>
<td>GRO-Seq</td>
<td>De novo transcript identification from strand specific GRO-Seq.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">DNase</span></code></td>
<td>DNase-Seq</td>
<td>Adjusted parameters for DNase-Seq peak finding.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">mC</span></code></td>
<td>MeDIP-Seq</td>
<td>DNA methylation analysis.</td>
</tr>
</tbody>
</table>
<p>Please select a style option that meets your analysis needs.</p>
<ol class="arabic simple" start="3">
<li>Add information about each sample in the peak calling table.</li>
<li>Click the Run button in the peak calling table.</li>
</ol>
</div>
</div>
<div class="section" id="graph">
<h2>5-5.Graph<a class="headerlink" href="#graph" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id6">
<h3>Quick Start<a class="headerlink" href="#id6" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>Select Analysis -&gt; Graph in the Menu bar.</li>
<li>Select the resulting folder generated by analyzing public data or private data.</li>
<li>set the TSS region and Bin size in the Graph table.</li>
<li>Click the Run button in the Graph table.</li>
</ol>
</div>
</div>
<div class="section" id="start-analyzing-graph">
<h3>Start analyzing Graph<a class="headerlink" href="#start-analyzing-graph" title="Permalink to this headline">¶</a></h3>
<img alt="_images/9.Graph_Start.png" src="_images/9.Graph_Start.png" />
<p>The Graph process is a step that extracts data for a specific region given by the user in the visualization file (bigWig format) and draws the heatmap and line plot using the extracted values.</p>
<p>Because the graph process uses the bigWig format file to calculate the value for a given region(bed format file) by the user, you must complete the processing of the previous steps for at least one sample.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Previous</span> <span class="pre">steps</span></code> : Public data or Private data analysis -&gt; Peak Calling.</li>
</ul>
<ol class="arabic simple">
<li>Select Analysis -&gt; Graph in the Menu bar.</li>
<li>Select the resulting folder generated by analyzing public data or private data.</li>
</ol>
</div>
<div class="section" id="graph-table">
<h3>Graph table<a class="headerlink" href="#graph-table" title="Permalink to this headline">¶</a></h3>
<img alt="_images/10.Graph_Table.png" src="_images/10.Graph_Table.png" />
<p>To draw graphs, Octopus-toolkit requires a bigWig format file and a bed format file.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">bigWig</span></code> : Output of the Public data or Private data analysis.</li>
<li><code class="docutils literal"><span class="pre">bed</span></code> : Output of the Peak calling.</li>
</ul>
<p>In the loaded data, bed format file is saved annotation option, and bigWig format file is saved in sample window.</p>
<p>To use Graph, you select a specific region to be analyzed in the annotation information for calculating value. And then, select the sample you want to analyze in the sample window and click the insert button.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Annotation(bed)</span></code></li>
</ul>
<p>The annotation (bed) option can be selected from the Promoter.bed provided by Octopus-toolkit and the bed file analyzed by the user in the peak calling process.</p>
<p>The Table option gives you a choice of additional options to calculate a specific region using the bigWig format file.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">TSS</span> <span class="pre">Region</span></code></li>
</ul>
<p>This option is used to calculate the distance of a regularly-sized from the locus of the selected annotation file.You can select the distance between upstream and downstream from a specific region.
dd
The unit for this option is basepair.(BP)</p>
<img alt="_images/14.TSS-Region.png" src="_images/14.TSS-Region.png" />
<p>The TSS-regions that the Octopus-toolkit provides to you are 1000,2000,5000 and 10000.</p>
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Number</span> <span class="pre">of</span> <span class="pre">BINs</span></code></li>
</ul>
<p>The region selected in the TSS region option is divided into n bin and the region is averaged. By selecting the Bin value, Octopus-toolkit automatically calculates the size of each region.</p>
<p>If the value of the selected Bin value is large, you can draw a smoother line plot.</p>
<p>The Number of BINs that the Octopus-toolkit provides to you are 50,100 and 200.</p>
<ol class="arabic simple" start="3">
<li>set the TSS region and Bin size in the Graph table.</li>
<li>Click the Run button in the Graph table.</li>
</ol>
</div>
</div>
<div class="section" id="visualization">
<h2>5-6.Visualization<a class="headerlink" href="#visualization" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id7">
<h3>Quick Start<a class="headerlink" href="#id7" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>Select Analysis -&gt; IGV in the Menu bar.</li>
<li>Select the resulting folder generated by analyzing public data or private data.</li>
<li>In the sample window, select the sample you want to analyze and click insert</li>
<li>Check that all genomes match in the IGV table.</li>
<li>If the genomes don&#8217;t match, select the genome to be analyzed from the table option.</li>
<li>Click the Run button.</li>
</ol>
</div>
</div>
<div class="section" id="start-analyzing-igv">
<h3>Start analyzing IGV<a class="headerlink" href="#start-analyzing-igv" title="Permalink to this headline">¶</a></h3>
<img alt="_images/10.IGV_Start.png" src="_images/10.IGV_Start.png" />
<p>The IGV function is a process of visualizing analyzed data through IGV, a visualization analysis tool.</p>
<p>IGV uses the bigWig format file of the analyzed sample. Therefore, in order to use IGV function, you must complete at least one sample of the data analysis. (Public or Private)</p>
<ol class="arabic simple">
<li>Select Analysis -&gt; IGV in the Menu bar.</li>
<li>Select the resulting folder generated by analyzing public data or private data.</li>
</ol>
</div>
<div class="section" id="igv-table">
<h3>IGV table<a class="headerlink" href="#igv-table" title="Permalink to this headline">¶</a></h3>
<img alt="_images/11.IGV_Table.png" src="_images/11.IGV_Table.png" />
<ul class="simple">
<li><code class="docutils literal"><span class="pre">Genome</span></code></li>
</ul>
<p>Genome information shows the genome of the samples inserted into the IGV table.</p>
<p>First, select the sample to be visualized using IGV and insert it into the IGV table.
So, If a different genome is added as shown in the table above, you must select a reference genome among several genomes.</p>
<ol class="arabic simple" start="3">
<li>In the sample window, select the sample you want to analyze and click insert</li>
<li>Check that all genomes match in the IGV table.</li>
<li>If the genomes don’t match, select the genome to be analyzed from the table option.</li>
<li>Click the Run button.</li>
</ol>
</div>
<div class="section" id="id8">
<h3>Run<a class="headerlink" href="#id8" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>Run screen of IGV.</li>
</ul>
<img alt="_images/12.IGV_Run.png" src="_images/12.IGV_Run.png" />
<p>Unlike other functions in Octopus-toolkit, the IGV tool runs separately from Octopus-toolkit. You can upload more data directly from the IGV, and can set IGV&#8217;s options.</p>
</div>
</div>
<div class="section" id="output">
<span id="id9"></span><h2>5-7.Output<a class="headerlink" href="#output" title="Permalink to this headline">¶</a></h2>
<p>The output file generated by each process is shown below.</p>
<img alt="_images/15.Process_output.png" src="_images/15.Process_output.png" />
<ul class="simple">
<li>In the result folder</li>
</ul>
<img alt="_images/16.Result_in_folder.png" src="_images/16.Result_in_folder.png" />
<table border="1" class="colwidths-given docutils">
<colgroup>
<col width="25%" />
<col width="25%" />
<col width="25%" />
<col width="25%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Folder name</th>
<th class="head">Process</th>
<th class="head">File format</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><code class="docutils literal"><span class="pre">00_Fastq</span></code></td>
<td>Preprocessing,Trimming</td>
<td><code class="docutils literal"><span class="pre">fastq</span></code>, <code class="docutils literal"><span class="pre">Trim.fastq</span></code></td>
<td>Save the raw read file and trimmed raw read file.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">00_SRA</span></code></td>
<td>Preprocessing</td>
<td><code class="docutils literal"><span class="pre">sra</span></code></td>
<td>Save compressed raw read file downloaded from NCBi</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">01_Fastqc</span></code></td>
<td>Quality check</td>
<td><code class="docutils literal"><span class="pre">html</span></code>, <code class="docutils literal"><span class="pre">txt</span></code></td>
<td>Save the result of the Quality check.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">02_Bam</span></code></td>
<td>Alignment</td>
<td><code class="docutils literal"><span class="pre">bam</span></code>, <code class="docutils literal"><span class="pre">sorted.bam</span></code>, <code class="docutils literal"><span class="pre">bai</span></code></td>
<td>Save the result data of Alignment and sorting.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">03_RNA_RPKM_Count</span></code></td>
<td>Normalization</td>
<td><code class="docutils literal"><span class="pre">RPKM</span></code>, <code class="docutils literal"><span class="pre">Count</span></code></td>
<td>Save the calculated RPKM and read count for the RNA-Seq data.</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">03_Tag</span></code></td>
<td>Visualization</td>
<td><code class="docutils literal"><span class="pre">Tag</span> <span class="pre">folder</span></code></td>
<td>Save the Tag folders created by Homer tool.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">04_BigWig</span></code></td>
<td>Visualization</td>
<td><code class="docutils literal"><span class="pre">bigWig</span></code></td>
<td>Save the result of the Visualization</td>
</tr>
<tr class="row-odd"><td><code class="docutils literal"><span class="pre">05_Analysis</span></code></td>
<td>Peak Calling,Annotation</td>
<td><code class="docutils literal"><span class="pre">bed</span></code>, <code class="docutils literal"><span class="pre">annotation</span></code></td>
<td>Save the Bed and Annotation files.</td>
</tr>
<tr class="row-even"><td><code class="docutils literal"><span class="pre">GSE57617.txt</span></code></td>
<td>Preprocessing~Visualization</td>
<td><code class="docutils literal"><span class="pre">txt</span></code></td>
<td>Sample.txt is a file that stores the analysis status and information of the data.</td>
</tr>
</tbody>
</table>
<div class="section" id="fastq">
<h3>00_Fastq<a class="headerlink" href="#fastq" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="full-parameter">
<span id="id10"></span><h2>5-8.Full Parameter<a class="headerlink" href="#full-parameter" title="Permalink to this headline">¶</a></h2>
<p>Full parameter provides options for 3rd party tools for analyzing NGS data in Octopus-toolkit.</p>
<p>The 3rd party tools used in Octopus-toolkit : <a class="reference internal" href="#rd-party-tool"><span class="std std-ref">3rd party tools</span></a></p>
<div class="section" id="preprocessing">
<h3>Preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this headline">¶</a></h3>
<p>Preprocessing is the process of downloading NGS data from NCBI or converting SRA format file to Fastq format file.
The 3rd party tools used in preprocessing are Aspera and SRAToolkit(fastq-dump)</p>
<ul>
<li><p class="first"><strong>Transfer rate</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">MAX-RATE</span></code> : MAX transfer rate (Only Integer)</p>
<p><code class="docutils literal"><span class="pre">MIN-RATE</span></code> : MIN transfer rate (Only Integer)</p>
<p><code class="docutils literal"><span class="pre">Overwrite</span></code> : Overwrite-Method, Always(Default), Never, Older, Diff</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Convert Sra to Fastq (Filtering)</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">MIN-Read</span> <span class="pre">Length</span></code> : Filter by sequence length &gt;= &lt;Value&gt; (Only Integer)</p>
<p><code class="docutils literal"><span class="pre">Aligned</span> <span class="pre">or</span> <span class="pre">unaligned</span> <span class="pre">reads</span></code> : Dump only aligned sequence or unaligned sequences, NotUse(Default), Both, Aligned, Unaligned</p>
<p><code class="docutils literal"><span class="pre">Quality</span> <span class="pre">conversion</span></code> (offset) : Offset to use for quality conversion, 33(Default), 64</p>
<p><code class="docutils literal"><span class="pre">Dump</span> <span class="pre">biological</span> <span class="pre">reads(Only)</span></code> : Dump only biological reads, No(Default)</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="qc-trimming">
<h3>QC &amp; Trimming<a class="headerlink" href="#qc-trimming" title="Permalink to this headline">¶</a></h3>
<p>QC &amp; Trimming is the process of measuring the quality of the reads and trimming the adapter sequence and low-quality reads.
The 3rd party tools used in QC &amp; Trimming are FastQC and Trimmomatic.</p>
<ul>
<li><p class="first"><strong>Determined quality of DNA Sequence</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">K-Mer</span></code> : Specifies the length of Kmer to look for in the Kmer content module, Specified Kmer length must be between 2 and 10. Default length is 7 if not specified.</p>
<p><code class="docutils literal"><span class="pre">Allocated</span> <span class="pre">memory</span></code> : Set the momory available on your computer for Quality check. Provides a measure of currently available memory . (Octopus-toolkit option)</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Trimmed DNA sequence data</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Illumina</span> <span class="pre">adapt</span> <span class="pre">Sequence</span></code> : Cut adapter and other illumina-specific sequences from the read.</p>
<p><code class="docutils literal"><span class="pre">Seed</span> <span class="pre">mismatches</span></code> : Specifies the maximum mismatch count which will still allow a full match to be performed</p>
<p><code class="docutils literal"><span class="pre">Palindrome</span> <span class="pre">clip</span> <span class="pre">threshold</span></code> : Specifies how accurate the match between the two &#8216;adapter ligated&#8217; reads must be for PE palindrome read alignment.</p>
<p><code class="docutils literal"><span class="pre">Simple</span> <span class="pre">clip</span> <span class="pre">threshold</span></code> : Specifies how accurate the match between any adapter etc. sequence must be against a read.</p>
<p><code class="docutils literal"><span class="pre">Window</span> <span class="pre">size</span></code> : specifies the number of bases to average across</p>
<p><code class="docutils literal"><span class="pre">Average</span> <span class="pre">quality</span></code> : Specifies the average quality required.</p>
<p><code class="docutils literal"><span class="pre">LEADING</span></code> : Specifies the minimum quality required to keep a base.</p>
<p><code class="docutils literal"><span class="pre">TRAILING</span></code> : Specifies the minimum quality required to keep a base.</p>
<p><code class="docutils literal"><span class="pre">HEADCROP</span></code> : The number of bases to keep, from the start of the read.</p>
<p><code class="docutils literal"><span class="pre">TAILCROP</span></code> : The number of bases to remove from the start of the read.</p>
<p><code class="docutils literal"><span class="pre">Minimum</span> <span class="pre">length</span> <span class="pre">of</span> <span class="pre">reads</span> <span class="pre">to</span> <span class="pre">be</span> <span class="pre">kept</span></code> : Specifies the minimum length of reads to be kept.</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="alignment">
<h3>Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a></h3>
<p>Alignment is the process of mapping reads to the reference genome.
The 3rd party tool used in Alignment is Hisat2.</p>
<ul>
<li><p class="first"><strong>Input</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Skip</span> <span class="pre">N</span> <span class="pre">read</span></code> : Skip the first &lt;int&gt; reads/pairs in the input (none)</p>
<p><code class="docutils literal"><span class="pre">Stop</span> <span class="pre">after</span> <span class="pre">aligning</span> <span class="pre">N</span> <span class="pre">reads</span></code> : Stop after first &lt;int&gt; reads/pairs (no limit)</p>
<p><code class="docutils literal"><span class="pre">Trim</span> <span class="pre">N</span> <span class="pre">bases</span> <span class="pre">5'</span> <span class="pre">end</span></code> : Trim &lt;int&gt; bases from 5&#8217;/left end of reads (0)</p>
<p><code class="docutils literal"><span class="pre">Trim</span> <span class="pre">N</span> <span class="pre">bases</span> <span class="pre">3'</span> <span class="pre">end</span></code> : Trim &lt;int&gt; bases from 3&#8217;/right end of reads (0)</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Scoring</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Ambiguous</span> <span class="pre">read</span> <span class="pre">penalty</span></code> : Penalty for non-A/C/G/Ts in read/ref</p>
<p><code class="docutils literal"><span class="pre">Mismatch</span> <span class="pre">penalty</span></code> : Max and min penalties for mismatch; lower qual = lower penalty &lt;2,6&gt;</p>
<p><code class="docutils literal"><span class="pre">Soft-Clipping</span> <span class="pre">penalty</span></code> : Max and min penalties for soft-clipping; lower qual = lower penalty &lt;1,2&gt;</p>
<p><code class="docutils literal"><span class="pre">Read</span> <span class="pre">gap</span> <span class="pre">penalty</span></code> : Read gap open, extend penalties (5,3)</p>
<p><code class="docutils literal"><span class="pre">Reference</span> <span class="pre">gap</span> <span class="pre">penalty</span></code> : Reference gap open, extend penalties (5,3)</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Alignment</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Ignore</span> <span class="pre">all</span> <span class="pre">quality</span> <span class="pre">values</span></code> : Treat all quality values as 30 on Phred scale (no)</p>
<p><code class="docutils literal"><span class="pre">Do</span> <span class="pre">not</span> <span class="pre">align</span> <span class="pre">reverse</span> <span class="pre">of</span> <span class="pre">read</span></code> : Do not align forward (original) version of read (no)</p>
<p><code class="docutils literal"><span class="pre">Do</span> <span class="pre">not</span> <span class="pre">align</span> <span class="pre">forward</span> <span class="pre">of</span> <span class="pre">read</span></code> : Do not align reverse-complement version of read (no)</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Spliced alignment</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Do</span> <span class="pre">not</span> <span class="pre">spliced</span> <span class="pre">alignment</span></code> : Disable spliced alignment</p>
<p><code class="docutils literal"><span class="pre">Canonical</span></code> : Penalty for a canonical splice site (0)</p>
<p><code class="docutils literal"><span class="pre">Non-canonical</span></code> : Penalty for a non-canonical splice site (12)</p>
<p><code class="docutils literal"><span class="pre">MIN-Length</span></code> : Minimum intron length (20)</p>
<p><code class="docutils literal"><span class="pre">MAX-Length</span></code> : Maximum intron length (500000)</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="visualization-tagdirectory">
<h3>Visualization-TagDirectory<a class="headerlink" href="#visualization-tagdirectory" title="Permalink to this headline">¶</a></h3>
<p>To analyze data using Homer, you need to make all useful information about the sample into the Tag directory.
Visualization-TagDirectory is the process of creating this Tag Directory.
The 3rd party tool used in TagDirectory is Homer.</p>
<ul>
<li><p class="first"><strong>Create tag directory</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Fragment-Length</span></code> : (Set estimated fragment length - given: use read lengths), By default treats the sample as a single read ChIP-Seq experiment</p>
<p><code class="docutils literal"><span class="pre">Maximum</span> <span class="pre">tags</span> <span class="pre">per</span> <span class="pre">bp</span></code> : Maximum tags per bp, default: no maximum</p>
<p><code class="docutils literal"><span class="pre">Flip</span> <span class="pre">the</span> <span class="pre">strands</span> <span class="pre">of</span> <span class="pre">each</span> <span class="pre">read</span></code> : Flip strand of each read, i.e. might want to use with some RNA-seq</p>
<p><code class="docutils literal"><span class="pre">Length</span> <span class="pre">of</span> <span class="pre">the</span> <span class="pre">read</span> <span class="pre">to</span> <span class="pre">keep</span></code> : Filter reads with lengths outside this range</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="visualization-makebigwig">
<h3>Visualization-MakeBigWig<a class="headerlink" href="#visualization-makebigwig" title="Permalink to this headline">¶</a></h3>
<p>MakeBigWig is the process of creating bigWig format file which is Visualization file using TagDirectory.
The 3rd party tool used in MakeBigWig is Homer.</p>
<ul>
<li><p class="first"><strong>Make visualization data</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Size</span> <span class="pre">of</span> <span class="pre">the</span> <span class="pre">bigWig</span> <span class="pre">files</span></code> : Size of file, when gzipped, default: 1e10, i.e. no reduction</p>
<p><code class="docutils literal"><span class="pre">Fragment</span> <span class="pre">Length</span></code> : Approximate fragment length, default: auto</p>
<p><code class="docutils literal"><span class="pre">Resolution</span></code> : Resolution, in bp, of file, default: 1, <code class="docutils literal"><span class="pre">avg</span></code> report average coverage if resolution is larger than 1bp, default: max is reported</p>
<p><code class="docutils literal"><span class="pre">Tags</span> <span class="pre">per</span> <span class="pre">bp</span> <span class="pre">to</span> <span class="pre">count</span></code> : Minimum and maximum tags per bp to count, default: no limit</p>
<p><code class="docutils literal"><span class="pre">Plot</span> <span class="pre">negative</span> <span class="pre">values</span></code> : Plot negative values, i.e. for - strand transcription</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Normalization</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Normalize</span> <span class="pre">the</span> <span class="pre">total</span> <span class="pre">number</span> <span class="pre">of</span> <span class="pre">reads</span></code> : Total number of tags to normalize experiment to, default: 1e7</p>
<p><code class="docutils literal"><span class="pre">Set</span> <span class="pre">the</span> <span class="pre">standard</span> <span class="pre">length</span></code> : Expected length of fragment to normalize to [0=off], default: 100</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="peakcalling-chip-seq-histone">
<h3>PeakCalling-ChIP-Seq/Histone<a class="headerlink" href="#peakcalling-chip-seq-histone" title="Permalink to this headline">¶</a></h3>
<p>PeakCalling is the process of detecting the region of the mapped read to the genome.
The 3rd party tool used in PeakCalling is Homer.</p>
<ul>
<li><p class="first"><strong>ChIP-Seq/Histone</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Peak</span> <span class="pre">size</span></code> : Peak size, default: 0</p>
<p><code class="docutils literal"><span class="pre">MIN-Distance</span></code> : Minimum distance between peaks, default: 0 (peak size x2)</p>
<p><code class="docutils literal"><span class="pre">Genome</span> <span class="pre">Size</span></code> : Set effective mappable genome size, default: 2e9</p>
<p><code class="docutils literal"><span class="pre">Fragment</span> <span class="pre">Length</span></code> : Approximate fragment length, default: auto</p>
<p><code class="docutils literal"><span class="pre">Input</span> <span class="pre">Fragment</span> <span class="pre">Length</span></code> : Approximate fragment length of input tags, default: auto</p>
<p><code class="docutils literal"><span class="pre">Tag</span></code> : Maximum tags per bp to count, 0 = no limit, default: auto</p>
<p><code class="docutils literal"><span class="pre">Input</span> <span class="pre">tag</span></code> : Maximum tags per bp to count in input, 0 = no limit, default: auto</p>
<p><code class="docutils literal"><span class="pre">Tag</span> <span class="pre">count</span> <span class="pre">to</span> <span class="pre">normalize</span></code> : Tag count to normalize to, default 10000000</p>
<p><code class="docutils literal"><span class="pre">Region</span> <span class="pre">Resolution</span></code> : Extends start/stop coordinates to cover full region considered &#8220;enriched&#8221; (YES), <code class="docutils literal"><span class="pre">Resolution</span></code> number of fractions peaks are divided in when extending &#8216;regions&#8217;, def: 4</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="peakcalling-peak-filter">
<h3>PeakCalling-Peak Filter<a class="headerlink" href="#peakcalling-peak-filter" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first"><strong>Peak Filter</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Fold</span> <span class="pre">Enrichment(Input)</span></code> : Fold enrichment over input tag count, default: 4.0</p>
<p><code class="docutils literal"><span class="pre">Poisson</span> <span class="pre">p-value</span> <span class="pre">threshold(Input)</span></code> : Poisson p-value threshold relative to input tag count, default: 0.0001</p>
<p><code class="docutils literal"><span class="pre">Fold</span> <span class="pre">Enrichment(Local)</span></code> : Fold enrichment over local tag count, default: 4.0</p>
<p><code class="docutils literal"><span class="pre">Poisson</span> <span class="pre">p-value</span> <span class="pre">threshold(Local)</span></code> : Poisson p-value threshold relative to local tag count, default: 0.0001</p>
<p><code class="docutils literal"><span class="pre">Fold</span> <span class="pre">Enrichment(Unique</span> <span class="pre">Tag)</span></code> : Fold enrichment limit of expected unique tag positions, default: 2.0</p>
<p><code class="docutils literal"><span class="pre">Local</span> <span class="pre">Size(Local</span> <span class="pre">tag)</span></code> : Region to check for local tag enrichment, default: 10000</p>
<p><code class="docutils literal"><span class="pre">Input</span> <span class="pre">Size(Input</span> <span class="pre">tag)</span></code> : Size of region to search for control tags, default: 0</p>
<p><code class="docutils literal"><span class="pre">`False</span> <span class="pre">Discovery</span> <span class="pre">Rate</span></code> : False discovery rate, default = 0.001</p>
<p><code class="docutils literal"><span class="pre">Poisson</span> <span class="pre">p-value</span> <span class="pre">cutoff</span></code> : Set poisson p-value cutoff, default: 0.001</p>
<p><code class="docutils literal"><span class="pre">Set</span> <span class="pre">#</span> <span class="pre">of</span> <span class="pre">tags</span></code> : Set # of tags to define a peak, default: 25</p>
<p><code class="docutils literal"><span class="pre">Set</span> <span class="pre">#</span> <span class="pre">of</span> <span class="pre">normalized</span> <span class="pre">tags</span></code> : Set # of normalized tags to define a peak, by default uses 1e7 for norm</p>
</div></blockquote>
</li>
</ul>
</div>
<div class="section" id="peakcalling-other-analysis">
<h3>PeakCalling-Other analysis<a class="headerlink" href="#peakcalling-other-analysis" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first"><strong>MethylC-Seq/BS-Seq</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">Find</span> <span class="pre">Region</span></code> : Find unmethylated/methylated regions, default: -unmethyC</p>
<p><code class="docutils literal"><span class="pre">Methyl</span> <span class="pre">Threshold</span></code> : Methylation threshold of regions, default: avg methylation/2</p>
<p><code class="docutils literal"><span class="pre">Min</span> <span class="pre">cytosine</span> <span class="pre">per</span> <span class="pre">Methyl</span></code> : Minimum number of cytosines per methylation peak, default: 6</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Gro-Seq</strong></p>
<blockquote>
<div><p><code class="docutils literal"><span class="pre">TSS</span> <span class="pre">Size</span></code> : Size of region for initiation detection/artifact size, default: 250</p>
<p><code class="docutils literal"><span class="pre">Min-body</span> <span class="pre">Size</span></code> : Size of regoin for transcript body detection, default: 1000</p>
<p><code class="docutils literal"><span class="pre">TSS-fold</span> <span class="pre">enrichment</span></code> : Fold enrichment for new initiation dectection, default: 4.0</p>
<p><code class="docutils literal"><span class="pre">Body-fold</span> <span class="pre">enrichment</span></code> : Fold enrichment for new transcript dectection, default: 4.0</p>
<p><code class="docutils literal"><span class="pre">End-fold</span> <span class="pre">enrichment</span></code> : End transcript when levels are this much less than the start, default: 10.0</p>
<p><code class="docutils literal"><span class="pre">Fragment</span> <span class="pre">Length</span></code> : Approximate fragment length, default: 150</p>
<p><code class="docutils literal"><span class="pre">Confidence</span> <span class="pre">P-value</span></code> : Confidence p-value: 1.00e-05</p>
<p><code class="docutils literal"><span class="pre">Pseudo</span> <span class="pre">Tag</span> <span class="pre">Count</span></code> : Pseudo tag count, default: 2.0</p>
<p><code class="docutils literal"><span class="pre">Minimum</span> <span class="pre">initial</span> <span class="pre">read</span> <span class="pre">depth</span></code> : Minimum initial read depth for transcripts, default: auto</p>
</div></blockquote>
</li>
</ul>
</div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="tutorial.html" class="btn btn-neutral float-right" title="6.Tutorial" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="user_interface.html" class="btn btn-neutral" title="4.User Interface" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016-2017, Keunsoo Kang, Taemook Kim.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'stable',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>